### PR TITLE
Docs - RTKQ - Flesh out content on hooks api page

### PR DESCRIPTION
**Pull request by Shrugsy**
Fleshes out missing content for the RTK Query hooks api page. 

- Adds pseudo type signatures for each hook
- Adds brief descriptions of hook parameters/return values
- Adds brief hook descriptions
- Adds 'feature' points for each hook
- Adds a 'feature comparison' table for a quick eyeball comparison of what each hook does

Some descriptions do feel a bit brief, but it's more than nothing

### CODE REVIEW

**Comment by Shrugsy** [comment]
@markerikson, @phryneas, @msutkowski; If anyone has some good descriptions for the missing/brief ones, please feel free to fill them in or post them here.

In particular, I wasn't confident in describing these accurately:
- `useQueryState` (description)
- `useQuerySubscription` (description)
- `useLazyQuerySubscription` (options + description)

**Comment by Shrugsy** [comment]
> @markerikson, @phryneas, @msutkowski; If anyone has some good descriptions for the missing/brief ones, please feel free to fill them in or post them here.
> 
> In particular, I wasn't confident in describing these accurately:
> ~- `useQueryState` (description)~
> ~- `useQuerySubscription` (description)~
> 
> * `useLazyQuerySubscription` (options + description)

I've added short descriptions for `useQueryState` and `useQuerySubscription`.

However I'm not sure exactly what `useLazyQuerySubscription` does.

If a `QuerySubscription` hook is concerned with the automatic fetching lifecycle, and the `Lazy` hook is concerned with opting out of the automatic fetching lifecycle to perform fetches manually, what does `useLazyQuerySubscription` do?

I'm sure I'm just missing a key understanding for one or both of the hooks

**Comment by markerikson** [comment]
My vague understanding just based on the chats and PRs is that the pieces are:

- "state": reads the data and re-renders
- "subscription": actually triggers a fetch of the data, and does so automatically
- "lazy": triggers a fetch of the data, but only when you tell it to manually
- "query": wraps matching "state" + "subscription" hooks

So, I _assume_ that `useLazyQuerySubscription` is "just a fetch of the data without reading it and re-rendering, but only triggers when you tell it to".

**Comment by Shrugsy** [comment]
> So, I assume that useLazyQuerySubscription is "just a fetch of the data without reading it and re-rendering, but only triggers when you tell it to".

Ah that wording made it click for me, thanks heaps. I was too tied up trying to correlate `subscription` with 'automatic fetching'

**Comment by markerikson** [comment]
Y'know, when you put it that way, it's almost like "subscription" is the wrong word to use for those hooks.  What would be better? "Fetch"? "Request"?  For that matter, once you trigger a lazy subscription, does all the normal timing/polling/whatever stuff kick in from there on out?  If so, maybe "subscription" is fine.

**Comment by Shrugsy** [comment]
> For that matter, once you trigger a lazy subscription, does all the normal timing/polling/whatever stuff kick in from there on out? If so, maybe "subscription" is fine.

Yep, it's exactly like that. I think `subscription` is fitting. Getting all of that info together, `LazyQuerySubscription` definitely makes sense

**Comment by phryneas** [comment]
@markerikson @Shrugsy  "subscription" is exactly that, but on another level than you think: the *component* subscribes to a *cache value in the store* to keep it from getting collected.

So yes, `useQuerySubscription` fetches data etc, but a normal `dispatch(initiate)` would do that too - but then it keeps it in the store and "unsubscribes" on unmount, so it's more about the component lifecycle synchronisation thing.
Same for `useLazyQuery`.

**Comment by Shrugsy** [comment]
Thanks @phryneas, that's a good clarification. I think the updated hook descriptions should be pretty accurate now

File: src/query/react/buildHooks.ts [hunk 1]
```diff
@@ -112,6 +124,19 @@ interface UseQuerySubscriptionOptions extends SubscriptionOptions {
   refetchOnMountOrArgChange?: boolean | number
 }
 
+/**
+ * A React hook that automatically triggers fetches of data from an endpoint, and 'subscribes' the component to the cached data.
```
Comment (msutkowski) [inline src/query/react/buildHooks.ts:128]:
```suggestion
 * A React hook that automatically triggers a request for a given endpoint, and 'subscribes' the component to the cached data.
```

Comment (msutkowski) [reply src/query/react/buildHooks.ts:128]: I think this copy would be slightly more readable and matches more closely to the `UseMutation` language below.

File: docs/api/rtk-query/created-api/hooks.md [hunk 2]
```diff
@@ -78,87 +231,281 @@ type UseQueryResult<T> = {
...
+type UsePrefetch = (
+  endpointName: string,
+  options?: UsePrefetchOptions
+) => PrefetchCallback
+
+type UsePrefetchOptions =
+  | {
+      // If specified, only runs the query if the difference between `new Date()` and the last
+      // `fulfilledTimeStamp` is greater than the given value
+      ifOlderThan?: false | number
```
Comment (msutkowski) [inline docs/api/rtk-query/created-api/hooks.md:486]: Should we note that this value is in seconds here?

**Review by msutkowski — COMMENTED** [review]
This is awesome! Thanks for all of the work here. Only a question and one suggestion on my end :trophy:

File: docs/api/rtk-query/created-api/hooks.md (hunk 2 above)
Comment (Shrugsy) [reply docs/api/rtk-query/created-api/hooks.md:486]: Yep, good point. Done now

**Review by msutkowski — APPROVED** [review]
👍

**Pull request merged**